src/sqlfluff/utils/reflow/sequence.py

```diff
@@ -458,3 +565,262 @@ def respace(
             # Generate the fix to do the removal.
             embodied_fixes=fixes,
         )
+
+    @staticmethod
+    def _identify_rebreak_spans(
+        element_buffer: ReflowSequenceType, root_segment: BaseSegment
+    ) -> List[_RebreakSpan]:
+        spans: List[_RebreakSpan] = []
+        # We'll need at least two elements each side, so constrain
+        # our range accordingly.
+        for idx in range(2, len(element_buffer) - 2):
+            # Only evaluate blocks:
+            elem = element_buffer[idx]
+            # Only evaluate blocks
+            if not isinstance(elem, ReflowBlock):
+                continue
+            # Does the element itself have config? (The easy case)
+            if elem.line_position:
+                # Blocks should only have one segment so it's easy to pick it.
+                spans.append(
+                    _RebreakSpan(elem.segments[0], idx, idx, elem.line_position)
+                )
+            # Do any of it's parents have config, and are we at the start
+            # of them?
+            for key in elem.line_position_configs.keys():
+                seg_idx, length, _ = elem.depth_info.stack_positions[key]
+                # If we're not at the start of the segment, then pass.
+                if seg_idx != 0:
+                    continue
+                # Can we find the end?
+                for end_idx in range(idx, len(element_buffer) - 2):
+                    end_elem = element_buffer[end_idx]
+                    if not isinstance(end_elem, ReflowBlock):
+                        continue
+                    if end_elem.depth_info.stack_positions[key][0] == length - 1:
+                        # Found the end. Add it to the stack.
+                        # We reference the appropriate element from the parent stack.
+                        target_depth = elem.depth_info.stack_hashes.index(key)
+                        target = root_segment.path_to(element_buffer[idx].segments[0])[
+                            target_depth
+                        ].segment
+                        spans.append(
+                            _RebreakSpan(
+                                target, idx, end_idx, elem.line_position_configs[key]
+                            )
+                        )
+                        break
+                else:
+                    # If we find the start, but not the end, it's not a problem, but
+                    # we won't be rebreaking this span. This is important so that we
+                    # don't rebreak part of something without the context of what's
+                    # in the rest of it. Continue.
+                    continue
```

Comment:
Does the `continue` here really do anything? We're at the end of the loop body already.

Reply:
Fair point, I'll tidy this up and make it more explicit.

Reply:
Cool. This was more of a nitpick / readability thing. I had to look at the code a few times to be sure I wasn't missing something.

One really simple way to address this would be to change the `continue` to a `pass`, but maybe you have a better idea. 👍